### PR TITLE
Update server interaction and xml templates to current state

### PR DESCRIPTION
**Pull request by idegtiarov**

### CODE REVIEW

File: static/js/generic_checkbox.html [hunk 1]
```diff
@@ -28,8 +28,8 @@
         function getUrlVars() {
           var vars = {};
           var url = document.URL;
-          url = url.replace('%24', '&');
-          url = url.replace('%3D', '=');
+          url = url.replace(/%24/g, '&');
+          url = url.replace(/%3D/g, '=');
```
Comment (z4y4ts) [inline static/js/generic_checkbox.html]: Why switch to regex? Are they as efficient as strings?

File: static/js/server_interaction.js [hunk 2]
```diff
@@ -5,16 +5,20 @@
...
     edx_adapt.course_id = $(".xmodule_display", window.parent.document).attr('data-course-id');
 
     //Convert course_id taken from the Course page into a course string compatible with EdxAdapt. E.g:
     //course_id = course-v1:University+SS001+2017_N is transformed into University/SS001/2017_N
     edx_adapt.coursestr = edx_adapt.course_id.split(':')[1].replace('+', '/');
 
+    courserawurl = (window.location != window.parent.location) ? document.referrer : document.location.href;
+    part_url = courserawurl.replace(/%3A/g, ':').replace(/%2B/g, '+').split(':');
+    url_list = part_url[2].split('/');
+    edx_adapt.courseid = url_list[0] + ':' + url_list[2];
```
Comment (z4y4ts) [inline static/js/server_interaction.js]: Use camelCase

File: static/js/generic_checkbox.html (hunk 1 above)
Comment (idegtiarov) [inline static/js/generic_checkbox.html]: We could have more than one character in the url which should be replaced.

Comment (z4y4ts) [inline static/js/generic_checkbox.html]: Ok, makes sense.

File: static/js/server_interaction.js [hunk 3]
```diff
@@ -5,16 +5,20 @@
...
+    edx_adapt.server = 'https://edx-adapt-dev.raccoongang.com'
 
     edx_adapt.course_id = $(".xmodule_display", window.parent.document).attr('data-course-id');
 
     //Convert course_id taken from the Course page into a course string compatible with EdxAdapt. E.g:
     //course_id = course-v1:University+SS001+2017_N is transformed into University/SS001/2017_N
     edx_adapt.coursestr = edx_adapt.course_id.split(':')[1].replace('+', '/');
 
+    courserawurl = (window.location != window.parent.location) ? document.referrer : document.location.href;
+    partUrl = courserawurl.replace(/%3A/g, ':').replace(/%2B/g, '+').split(':');
```
Comment (idegtiarov) [inline static/js/server_interaction.js]: Really seems so, improved.

File: static/js/server_interaction.js [hunk 4]
```diff
@@ -151,7 +155,7 @@
 
                         if (data['next'] == null) {
                             //The server just hasn't set the next problem yet. Retry twice, wait between attempts for 1s.
-                            if (retry < 2) {
+                            if (retry < 1) {
```
Comment (idegtiarov) [inline static/js/server_interaction.js]: done

File: static/js/server_interaction.js [hunk 5]
```diff
@@ -5,15 +5,20 @@
 (function( edx_adapt, $, undefined) {
 
     //Connection information
-    edx_adapt.apistr    = '/api/v1';
-    edx_adapt.server    = 'https://edx-adapt-dev.raccoongang.com';
+    edx_adapt.apistr = '/api/v1';
+    edx_adapt.server = 'https://edx-adapt-dev.raccoongang.com'
```
Comment (z4y4ts) [inline static/js/server_interaction.js]: Missed semicolon

Comment (idegtiarov) [inline static/js/server_interaction.js]: fixed

**Pull request merged**